tests/unit.c

```diff
@@ -187,8 +187,46 @@ bool testfuse8() {
   return true;
 }
 
+bool testfuse8_small() {
```

Comment:
Note: this is a direct copy of `testfuse8`, only difference is `size` so if you do want to merge this PR later I can refactor this into a function taking a `size` parameter and test a few different sizes.

I just sent this as-is to see if I could get help debugging why this happens